detectable/src/main/java/com/synopsys/integration/detectable/detectable/util/PoetrySectionResult.java

```diff
@@ -0,0 +1,47 @@
+/**
+ * detectable
+ *
+ * Copyright (c) 2021 Synopsys, Inc.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.detectable.detectable.util;
+
+import java.util.Optional;
+
+import org.tomlj.TomlTable;
+
+public class PoetrySectionResult {
+    private final boolean found;
+    private final TomlTable poetrySection;
+
+    public static PoetrySectionResult SECTION_NOT_FOUND = new PoetrySectionResult(false, null);
+
+    public PoetrySectionResult(final boolean found, final TomlTable poetrySection) {
```

Comment:
Typically for these the Detect approach is private constructor with static methods. This way the caller of the constructor doesn't need to pass false.

`static FOUND(value) { return new PoetrySection(true, value); }` and `static NOT_FOUND() {return new PoetrySection(false, null);`